### PR TITLE
Add org units to polio forms

### PR DESCRIPTION
**Pull request by xavierbarbosa**
### Added
- Org units levels select box, that can request child org units recursively

### CODE REVIEW

File: plugins/polio/js/src/hooks/useGetAuthenticatedUser.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { useQuery } from 'react-query';
+import { sendRequest } from '../utils/networking';
+
+export const useGetAuthenticatedUser = () =>
+    useQuery(['polio', 'user'], () => sendRequest('GET', '/api/profiles/me'));
```
Comment (geoffreydhuyvetters) [inline plugins/polio/js/src/hooks/useGetAuthenticatedUser.js:5]: I would use `['profiles', 'me']` here

File: plugins/polio/js/src/hooks/useGetOrgUnits.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { useQuery } from 'react-query';
+import { sendRequest } from '../utils/networking';
+
+export const useGetOrgUnits = (parent, source) =>
+    useQuery(['polio', 'org_units', parent, source], () => {
```
Comment (geoffreydhuyvetters) [inline plugins/polio/js/src/hooks/useGetOrgUnits.js:5]: I would use an object with the arguments as the 3rd param here, as I shared in the channel

`['orgunits', {parent_id: 0, source: 1, validation_status=valid}]`

File: plugins/polio/js/src/hooks/useGetOrgUnits.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        const params = new URLSearchParams(
+            `parent_id=${parent}&validation_status=VALID`,
+        );
+
+        if (source !== undefined) {
+            params.set('source', source);
+        }
```
Comment (geoffreydhuyvetters) [inline plugins/polio/js/src/hooks/useGetOrgUnits.js:12]: You can also construct URLSearchParams from a record, might be a bit easier to follow

```
const params4 = new URLSearchParams({"foo": "1", "bar": "2"});

```

**Comment by geoffreydhuyvetters** [comment]
I think we accidentally deleted the package-lock here

**Comment by xavierbarbosa** [comment]
@geoffreydhuyvetters please give this another look!

File: plugins/polio/js/src/hooks/useGetOrgUnits.js [hunk 4]
```diff
@@ -8,31 +8,48 @@ export const useGetOrgUnits = (parent, source) => {
         validation_status: 'VALID',
     };
 
-    return useQuery(['org_units', params], () => {
-        const queryString = new URLSearchParams(params);
+    return useQuery(
+        ['org_units', params],
```
Comment (geoffreydhuyvetters) [inline plugins/polio/js/src/hooks/useGetOrgUnits.js:12]: can we mimic the API routes? makes it clearer to follow in the cache

File: plugins/polio/js/src/hooks/useGetOrgUnits.js [hunk 5]
```diff
@@ -8,31 +8,48 @@ export const useGetOrgUnits = (parent, source) => {
...
+    );
 };
 
 export const useGetAllParentsOrgUnits = initialOrgUnit => {
-    return useQuery(['org_units_parents', initialOrgUnit], async () => {
-        if (initialOrgUnit === 0) {
-            return [0];
-        }
+    return useQuery(
+        ['org_units_parents', initialOrgUnit],
```
Comment (geoffreydhuyvetters) [inline plugins/polio/js/src/hooks/useGetOrgUnits.js:33]: same here.

File: plugins/polio/js/src/hooks/useGetOrgUnits.js (hunk 4 above)
Comment (xavierbarbosa) [reply plugins/polio/js/src/hooks/useGetOrgUnits.js:12]: The params is a variable with the object with all the parameters of the URL.

**Review by geoffreydhuyvetters — APPROVED** [review]
🚢

**Pull request merged**